### PR TITLE
Add API endpoint

### LINKED ISSUE #8: Add API endpoint
Add a client endpoint  to communicate with our backend API.
All CRUD actions for todo lists and tasks should be implemented.

Requirements for API:
CRUD actions for todo list
CRUD actions for tasks

URL's can be found in the backend branch https://github.com/wesharehoodies/todo-react/blob/backend/api.js

### PR DESCRIPTION
**Pull request by CasiOo**
Added a generic HttpClient, and the endpoints needed to communicate with out backend.
See https://github.com/wesharehoodies/todo-react/issues/8 for requirements

@arreche there were some comments you made in the last commit, that have not been resolved. They aren't deal breakers though.

@arreche  and @wesharehoodies please give your final thoughts.

Closes #8

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -21,4 +21,14 @@ module.exports = {
       }
     ],
   },
+  resolve : {
+    alias: {
+        jquery: __dirname + '/node_modules/jquery/dist/jquery.js'
```
Comment (arreche) [inline webpack.config.js]: Did you forget to save the `jquery` dependency in `package.json`?

File: index.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8"/>
+    
+  </head>
+  <body>
+    <div id="app">Placeholder</div>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
```
Comment (arreche) [inline index.html]: If `jquery` is being used from `node_modules` then this `script` can be removed

File: app/api/httpClient.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                data: queryParams,
+                success: resolve,
+                error: reject,
+                traditional: true
+            });
+        });
+    }
+    
+    POST(url, data) {
+        return new Promise((resolve, reject) => {
```
Comment (arreche) [inline app/api/httpClient.js]: The only different thing in those four code blocks (`GET`, `POST`, `PUT` and `DELETE`) is the `method` property.

* https://sourcemaking.com/refactoring/duplicated-code

File: webpack.config.js (hunk 1 above)
Comment (arreche) [inline webpack.config.js]: In a real world problem I would choose a more specific library for the requests instead

Comment (CasiOo) [inline webpack.config.js]: jquery is added as a dependency in `package.json`

File: index.html (hunk 2 above)
Comment (CasiOo) [inline index.html]: This index.html file should be removed, just like in master. Will fix :)

**Comment by arreche** [comment]
:+1:

**Pull request merged**